### PR TITLE
fix(tab): mdc-web v1.0.0 upgrade

### PR DESCRIPTION
**Pull request by gugu**

### CODE REVIEW

File: packages/tab-bar/index.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ class TabBar extends React.Component<
     this.foundation.init();
     const {activeIndex, indexInView} = this.props;
     if (this.tabList[activeIndex]) {
-      this.tabList[activeIndex].activate({} /* previousIndicatorClientRect */);
+      this.tabList[activeIndex].activate(new DOMRect() /* previousIndicatorClientRect */);
```
Comment (moog16) [inline packages/tab-bar/index.tsx]: This is actually not IE11 compatible. I also realize we currently don't work on IE11, but we have some issues to be IE11 compatible. Please just change this to:

```
// new DOMRect is not IE11 compatible
const defaultDOMRect = {
bottom: 0
height: 0
left: 0
right: 0
top: 0
width: 0
x: 0
y: 0
};
...activate(defaultDOMRect);
```

You may need to cast it.

File: packages/tab/index.tsx [hunk 2]
```diff
@@ -38,10 +38,11 @@ export interface TabProps extends React.HTMLProps<HTMLButtonElement> {
   isMinWidthIndicator?: boolean;
   stacked?: boolean;
   previousIndicatorClientRect?: ClientRect;
+  onInteration?: () => void;
 }
 
 interface MDCTabElementAttributes {
-  'aria-selected': boolean;
+  'aria-selected': 'false' | 'true' | undefined;
```
Comment (moog16) [inline packages/tab/index.tsx]: change to `'aria-selected'?: 'false' | 'true'`

File: packages/tab/index.tsx [hunk 3]
```diff
@@ -81,7 +82,7 @@ export default class Tab extends React.Component<TabProps, TabState> {
     const {active, focusOnActivate} = this.props;
     this.foundation = new MDCTabFoundation(this.adapter);
     this.foundation.init();
-    this.foundation.setFocusOnActivate(focusOnActivate);
+    this.foundation.setFocusOnActivate(focusOnActivate as boolean);
```
Comment (moog16) [inline packages/tab/index.tsx]: change this to `focusOnActivate!`. This has a defaultProp, so we can `!`.

File: packages/tab/index.tsx [hunk 4]
```diff
@@ -94,7 +95,7 @@ export default class Tab extends React.Component<TabProps, TabState> {
   componentDidUpdate(prevProps: TabProps) {
     const {active, focusOnActivate, previousIndicatorClientRect} = this.props;
     if (focusOnActivate !== prevProps.focusOnActivate) {
-      this.foundation.setFocusOnActivate(focusOnActivate);
+      this.foundation.setFocusOnActivate(focusOnActivate as boolean);
```
Comment (moog16) [inline packages/tab/index.tsx]: same with here.

File: packages/tab/index.tsx [hunk 5]
```diff
@@ -136,12 +137,15 @@ export default class Tab extends React.Component<TabProps, TabState> {
...
+          this.tabContentRef.current.offsetLeft :
+          0,
       getContentOffsetWidth: () =>
-        this.tabContentRef.current &&
-        this.tabContentRef.current.offsetWidth,
+        this.tabContentRef.current ?
+          this.tabContentRef.current.offsetWidth :
+          0,
       focus: () => this.tabRef.current && this.tabRef.current.focus(),
+      notifyInteracted: () => this.props.onInteration,
```
Comment (moog16) [inline packages/tab/index.tsx]: notifyInteracted: this.props.onInteraction

File: packages/tab/index.tsx [hunk 6]
```diff
@@ -38,10 +38,11 @@ export interface TabProps extends React.HTMLProps<HTMLButtonElement> {
   isMinWidthIndicator?: boolean;
   stacked?: boolean;
   previousIndicatorClientRect?: ClientRect;
+  onInteration?: () => void;
```
Comment (moog16) [inline packages/tab/index.tsx]: onInteraction

**Comment by moog16** [comment]
Screen Shot 2019-03-18 at 11 46 39 AM

Getting a warning. Please fix

**Comment by moog16** [comment]
@gugu  can you please also add the new prop to the readme

**Comment by gugu** [comment]
@moog16 done

**Pull request merged**